src/plugins/data_view_editor/public/components/data_view_editor_flyout_content.tsx

```diff
@@ -410,7 +410,12 @@ const IndexPatternEditorFlyoutContentComponent = ({
         </Form>
         <Footer
           onCancel={onCancel}
-          onSubmit={(adhoc?: boolean) => {
+          onSubmit={async (adhoc?: boolean) => {
+            const formData = form.getFormData();
+            if (!formData.name) {
+              form.updateFieldValues({ name: formData.title });
+              await form.getFields().name.validate();
+            }
             form.setFieldValue('isAdHoc', adhoc || false);
             form.submit();
```

Comment:
> 1. Reorder the fields so that "Index pattern" is first, for now
> 2. Set the placeholder value of the "Name" field to the index pattern value

+1 for this,  this would further improve the UX in my opinion, you would get an important info first: which indices would match your pattern, instead of having the first focus on the name field.